### DISCUSSION TITLE
antd 4.17.4 table 嵌套子表格联动失效

### DISCUSSION THREAD

**Discussion by wuxinheng** [ant-design/ant-design#50677] [Q&A] [answered]
这是我的代码示例：
看了官网文档，根据我自己的理解通过key 来进行关联，下面链接是我的代码实现
https://paste.ofcode.org/5nrLAZSKkaCPeyaWLBT9wC

我需要的效果：
点击一行，展开这一行的关联子数据

实际的效果：
点开一行，展开所有的子数据
![image](https://github.com/user-attachments/assets/32560461-b639-4dcb-bd71-32bcfec8f6db)

需要社区前辈指教一下 :sob:

**Comment by huntdream** [comment, marked as answer]
如果你的子数据都是在一个数组里面取，那么在`expandedRowRender `里面要对数据进行过滤的，你`data`里面传进去什么就会渲染什么，组件本身不会去处理

**Reply by wuxinheng** [reply]
感谢你的回答，我已经解决了这个问题
1.点击一行全部展开，是因为rowKey没有设置

**Reply by wuxinheng** [reply]
请教一下，我的列和数据是后端传给我的，我怎么做可以在这个基础上合并单元格
![image](https://github.com/user-attachments/assets/b7a7f6ed-52f2-4e0a-95e3-ebe0f8c3af5a)

**Reply by huntdream** [reply]
正常来说你在返回的columns里面把数据map一下，插入onCell就好了，不知道你的具体需求是啥

**Reply by wuxinheng** [reply]
具体需求是这样的
![image](https://github.com/user-attachments/assets/281f40f6-cae2-4b58-8f1a-dd50228b2dc1)
合并子表格中的物料编号和描述，我现在通过map 插入了oncell
这是我修改之后的代码
![image](https://github.com/user-attachments/assets/0a75ed1b-30dd-46e4-acf1-a4c96a78d284)
我的输出
![image](https://github.com/user-attachments/assets/c2f652e5-787f-4b33-a32f-818f80732e86)
现在没有效果

**Reply by huntdream** [reply]
既然onCell都插进去了，理论上应该没问题啊。
我用官方的例子改了一下，合并是正常的
![image](https://github.com/user-attachments/assets/fbca1c03-8ca9-46f1-83a5-478e75cefead)

**Reply by wuxinheng** [reply]
目前只找到了一个大概的方向
https://github.com/ant-design/pro-components/issues/8701

**Reply by huntdream** [reply]
你看能不能弄一个复现的codesandbox出来
